pkg/leader/leader.go

```diff
@@ -109,7 +109,33 @@ func Become(ctx context.Context, lockName string) error {
 			log.Info("Became the leader.")
 			return nil
 		case apierrors.IsAlreadyExists(err):
-			log.Info("Not the leader. Waiting.")
+			existing := &corev1.ConfigMap{}
+			key := crclient.ObjectKey{Namespace: ns, Name: lockName}
+			err := client.Get(ctx, key, existing)
+			if err != nil {
+				return err
+			}
+			leaderPod := &corev1.Pod{}
+			key = crclient.ObjectKey{Namespace: ns, Name: existing.ObjectMeta.OwnerReferences[0].Name}
+			err = client.Get(ctx, key, leaderPod)
+			if err != nil {
+				return err
+			}
+
+			podFailed := leaderPod.Status.Phase == corev1.PodFailed
+			podEvicted := leaderPod.Status.Reason == "Evicted"
+			if podFailed && podEvicted {
+				log.Info("Operator pod with leader lock has been evicted.", "Leader pod", leaderPod.Name)
+				err := client.Delete(ctx, leaderPod)
+				if err == nil {
+					log.Info("Evicted leader pod has been deleted.")
+				} else {
+					log.Error(err, "Evicted leader pod could not be deleted")
+
+				}
```

Comment:
If this pod successfully deletes the evicted leader pod, should it skip the backoff wait and try to immediately create the leader lock for itself?

Reply:
Since pod deletion can take time, I am letting the backoff loop avoid spamming delete requests. wdyt?

Reply:
Ah, yeah that makes sense. 👍